docs/source/process.rst

```diff
@@ -0,0 +1,570 @@
+    >>> shuffled_dataset = dataset.shuffle(buffer_size=10_000, seed=42)
+
+.. tip::
+
+    For datasets that are sharded into multiple files, :func:`datasets.IterableDataset.shuffle` will also shuffle the order of the shards.
+
+Reshuffle
+^^^^^^^^^
+
+Sometimes you may want to reshuffle the dataset after each epoch, so you will need to set a different seed. Use ``set_epoch()`` in between epochs to tell the dataset what epoch you're on. This way, the data will be shuffled using a seed of ``initial seed + current epoch``:
+
+    >>> for epoch in range(epochs):
+    ...     shuffled_dataset.set_epoch(epoch)
+    ...     for example in shuffled_dataset:
+    ...         ...
+
+Split dataset
+^^^^^^^^^^^^^
+
+You can split your dataset one of two ways: :func:`datasets.IterableDataset.take`, or :func:`datasets.IterableDataset.skip`.
+
+* :func:`datasets.IterableDataset.take` returns the first ``n`` examples in a dataset:
+
+    >>> dataset = load_dataset('oscar', "unshuffled_deduplicated_en", split='train', streaming=True)
+    >>> dataset_head = dataset.take(2)
+    >>> list(dataset_head)
+    [{'id': 0, 'text': 'Mtendere Village was...'}, '{id': 1, 'text': 'Lily James cannot fight the music...'}]
+
+* :func:`datasets.IterableDataset.skip` omits the first ``n`` examples in a dataset and returns the remaining examples:
+
+    >>> train_dataset = shuffled_dataset.skip(1000)
+
+.. important::
+
+    ``take`` and ``skip`` prevents future calls to ``shuffle`` because they lock in the order of the shards. You should ``shuffle`` your dataset before splitting it.
+
+.. _mix_label:
+Mix
+^^^
+
+Iterable Datasets can also be mixed together with :func:`datasets.interleave_datasets`. The mixed dataset returns alternating examples from each of the original datasets. 
+
+    >>> from datasets import interleave_datasets
+    >>> from itertools import islice
+    >>> en_dataset = load_dataset('oscar', "unshuffled_deduplicated_en", split='train', streaming=True)
+    >>> fr_dataset = load_dataset('oscar', "unshuffled_deduplicated_fr", split='train', streaming=True)
+    >>>
+    >>> multilingual_dataset = interleave_datasets([en_dataset, fr_dataset])
+    >>> print(list(islice(multilingual_dataset, 2)))
+    [{'text': 'Mtendere Village was inspired by the vision...}, {'text': "Média de débat d'idées, de culture et de littérature....}]
+
+For more control over how each of the original datasets are sampled and mixed, define sampling probabilities for each of the original datasets. In the following example, 80% of the final mixed dataset is in English and 20% is in French:
+
+    >>> multilingual_dataset_with_oversampling = interleave_datasets([en_dataset, fr_dataset], probabilities=[0.8, 0.2], seed=42)
+    >>> print(list(islice(multilingual_dataset_with_oversampling, 2)))
+    [{'text': 'Mtendere Village was inspired by the vision...}, {'text': 'Lily James cannot fight the music...}]
+
+Format
```

Comment:
From this point, all the following sections are only relevant for the regular map-style `Dataset` object, not `IterableDataset`

Reply:
Added an admonition alerting the user these methods only work for regular `Dataset` objects. This does make me lean more towards creating a separate page for processing an `IterableDataset` ;)